homeassistant/components/nut/__init__.py

```diff
@@ -190,29 +200,29 @@ def __init__(self, host, port, alias, username, password):
         # Establish client with persistent=False to open/close connection on
         # each update call.  This is more reliable with async.
         self._client = PyNUTClient(self._host, port, username, password, 5, False)
-        self.ups_list = None
-        self._status = None
-        self._device_info = None
+        self.ups_list: dict[str, str] | None = None
+        self._status: dict[str, str] | None = None
+        self._device_info: DeviceInfo | None = None
 
     @property
-    def status(self):
+    def status(self) -> dict[str, str] | None:
         """Get latest update if throttle allows. Return status."""
         return self._status
 
     @property
-    def name(self):
+    def name(self) -> str:
         """Return the name of the ups."""
-        return self._alias
+        return self._alias or f"Nut-{self._host}"
 
     @property
-    def device_info(self):
+    def device_info(self) -> DeviceInfo:
         """Return the device info for the ups."""
-        return self._device_info or {}
+        return self._device_info or DeviceInfo()
```

Comment:
Can we avoid returning empty DeviceInfo?

Reply:
You mean return `None` when nothing is available?